lang/rust/Config.in

```diff
@@ -0,0 +1,121 @@
+		bool "Build compiler documentation (--enable-compiler-docs)"
+		default n
+
+	config RUST_OPTIMIZE_TESTS
+		bool "Build tests with optimizations (--enable-optimized-tests)"
+		default n
+
+	config RUST_PARALLEL
+		bool "Build with multi-threaded support (--enable-parallel-compiler)"
+		default n
+
+	config RUST_VERBOSE_TESTS
+		bool "Enable verbose output when running tests (--enable-verbose-tests)"
+		default n
+
+	config RUST_CCACHE
+		bool "Build with ccache enabled (--enable-ccache)"
+		default n
+
+	config RUST_LLVM_STATIC
+		bool "Statically link to libstdc++ to LLVM (--enable-llvm-static-stdccp)"
+		default n
+
+	config RUST_LLVM_SHARED
+		bool "Prefer shared linking to LLVM (--enable-llvm-link-shared)"
+		default n
+
+	config RUST_CODEGEN_TESTS
+		bool "Run the src/test/codegen tests (--enable-codegen-tests)"
+		default n
+
+	config RUST_OPTION_CHECKING
+		bool "Complain about unrecognized options in this configure script (--enable-option-checking)"
+		default y
+
+	config RUST_ENABLE_NINJA
+		bool "Build LLVM using the Ninja generator (--enable-ninja)"
+		default y
+
+	config RUST_LOCKED_DEPS
+		bool "Force Cargo.lock to be up to date (--enable-locked-deps)"
+		default n
+
+	config RUST_VENDOR
+		bool "Enable usage of vendored Rust crates (--enable-vendor)"
+		default n
+
+	config RUST_SANITIZERS
+		bool "Build the sanitizer runtimes (asan, lsan, msan, tsan) (--enable-sanitizers)"
+		default n
+
+	config RUST_DIST_SRC
+		bool "When building tarballs enables building a source tarball (--enable-dist-src)"
+		default n
+
+	config RUST_CARGO_NATIVE_STATIC
+		bool "Build static native libraries in Cargo (--enable-cargo-native-static)"
+		default n
+
+	config RUST_PROFILER
+		bool "Build the profiler runtime (--enable-profiler)"
+		default n
+
+	config RUST_FULL_TOOLS
+		bool "Build all tools (--enable-full-tools)"
+		default y
+
+	config RUST_MISSING_TOOLS
+		bool "Allow failures when building tools (--enable-missing-tools)"
+		default y
+
+	config RUST_USE_LIBCXX
+		bool "Build LLMV with libc++ (--enable-use-libcxx)"
+		default n
+
+	config RUST_CONTROL_FLOW_GUARD
+		bool "Enable Control Flow Guard (--enable-control-flow-guard)"
+		default n
+
+	config RUST_OPTIMIZE
+		bool "Build optimized rust code (--enable-optimize)"
+		default n
+
+	config RUST_OPTIMIZE_LLVM
+		bool "Build optimized LLVM (--enable-optimize-llvm)"
+		default n
+
+	config RUST_LLVM_ASSERTIONS
+		bool "Build LLVM with assertions (--enable-llvm-assertions)"
+		default n
+
+	config RUST_DEBUG_ASSERTIONS
+		bool "Build with debugging assertions (--enable-debug-assertions)"
+		default n
+
+	config RUST_LLVM_RELEASE_DEBUGINFO
+		bool "Build LLVM with debugger metadata (--enable-llvm-release-debuginfo)"
+		default n
+
+	config RUST_MANAGE_SUBMODULES
+		bool "Let the build manage the git submodules (--enable-manage-submodules)"
+		default y
+
+	config RUST_FULL_BOOTSTRAP
+		bool "Full Bootstrap - Build three compilers instead of two (--enable-full-bootstrap)"
+		default y
```

Comment:
Why enable this by default? It shouldn't be needed and it will make the build take half again as long. https://github.com/rust-lang/rust/blob/9503ea19edbf01b9435e80e17d60ce1b88390116/config.toml.example#L243-L248

Reply:
It was my understanding that in order to get stage2 artifacts, the full bootstrap needed to happen (so it didn't stop at just stage0/1)?  Remember, almost everyone using Rust-lang in an OpenWrt context will be cross-compiling and will need the stage2 artifacts that would be skipped without the full bootstrap.  Did I misunderstand?

Reply:
I think x.py should automatically rebuild stage2 if it needs to for cross compiling. I would try that first to see if it works.

Also, I'm slightly confused why you would need 3 compilers in the first place, I would expect a stage 2 compiler to work fine. That's what's distributed by rustup.

Reply:
x.py only gets call ONCE, during the initial compile.  After that, the installation binaries I create are stored in an archive that is used instead of touching the rust source at all (unless the rust package version changes in the OpenWrt upstream).  This means I only run x.py a single time for an install..

`-rw-r--r-- 1 grommish grommish 520960752 Feb 11 00:57 rust-1.49.0-x86_64-unknown-linux-gnu_mips64-openwrt-linux-musl-install.tar.xz`

That is an example of my toolchain installation files (basically, a one-off version of rustup I guess?) that are built during the first compile. Subsequent runs, even on a `make clean`, sees that file, extracts it, runs the installs, and calls it a day.

Reply:
Seems reasonable, but I'm not sure how it's relevant. This is specifically about the rustc-dev component, which gets compiled with x.py at the same time as everything else. If you remove rustc-dev I think this option won't even be used either way.

Reply:
I'm testing removing the -dev options as we speak to see if it breaks anything.. then, if not, I'll test without the full bootstrap